CONFIGURATION.md

```diff
@@ -61,6 +61,26 @@ require('instana-nodejs-sensor')({
 });
 ```
 
+## Disabling Reporting Uncaught Exceptions
+The Instana Node.js sensor has the ability to report uncaught exceptions. By default, a Node.js process will be terminated by an uncaught exception (see [Node.js docs](https://nodejs.org/api/process.html#process_event_uncaughtexception)). If uncaught exception reporting is enabled, the Instana Node.js sensor will register a listener for the `uncaughtException` event and take the following actions when an uncaught exception occurs:
+
+* Report this as an incident to Instana, including the uncaught exception and its stack trace.
+* Finish the currently active span and mark it as an error (if automatic tracing is enabled).
+
+It will then rethrow the original exception to terminate the Node.js process. (Keeping a Node.js process running after an uncaught exception has occurred is strongly discouraged, as the process might be in an inconsistent state, see [Node.js docs](https://nodejs.org/api/process.html#process_warning_using_uncaughtexception_correctly).)
+
+A timeout of one second is applied, so if one or both of these actions have not completed after one second, the process will be terminated anyway.
```

Comment:
During this one second, will HTTP servers or message consumers continue to accept requests / consume messages? If so, I think we are influencing Node.js applications. Normally, message consumption would stop right away. The same is true for HTTP servers. If they don't stop consuming / accepting, then application code might run into inconsistent state and fail. This could, in the worst case, result in inconsistent / wrong data.

The only way to circumvent this that I could think of would be forcing synchronous HTTP requests to be used for data transmission to the agent. This will ensure that application code never gets any processing time in order to handle requests / messages.

Reply:
Very good point. I looked into doing synchronous requests briefly, I can look into that again more closely to see if it could be used here. I agree, if that's not possible we should definitely do this opt-in/disabled by default.

Reply:
@bripkens 

It seems the only possible option would be to use something like https://github.com/ForbesLindesay/sync-rpc, which does `child_process.spawnSync` to start a worker that does the asynchronous work for your main process. This is quite a hack and something I'd rather not do in an `uncaughtException` handler. My preferred route would be switch back to opt-in and more explicitly state the risks.

Thoughts?

Reply:
Another idea about not keeping the process alive and able to accept more requests - I think we might get a pretty good result when using core idea of sync-rpc (spawning another process) trimmed down to our actual needs. sync-rpc uses `spawnSync` to wait for the result of the asynchronous action. We actually do not care about the results of our two http requests, we can trigger them in a fire-and-forget style.

So, here's the new plan: Spawn a child process via `spawn` instead of `spawnSync`and then terminate the main process immediately. The child process can then take as much time as it wants to talk to the agent and the main process terminates synchronously in the `uncaughtException` handler.